### PR TITLE
[http-proxy-middleware] Set config as optional parameter

### PR DESCRIPTION
**Pull request by CelineBen**
## Description
The documentation of `http-proxy-middleware` mentions the use of shorthands where the only parameter is context or uri but the types did not allow for one string parameter.
I set the config parameter as optional

## Checklist 
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/chimurai/http-proxy-middleware#shorthand
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/http-proxy-middleware/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for http-proxy-middleware 0.19
+// Type definitions for http-proxy-middleware 0.20
```
Comment (BendingBender) [inline types/http-proxy-middleware/index.d.ts]: Why 0.20? Latest version of http-proxy-middleware is 0.19, as far as I can see.

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/http-proxy-middleware/index.d.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import * as httpProxy from "http-proxy";
 import { Readable } from 'stream';
 
 declare function proxy(config: proxy.Config): proxy.Proxy;
-declare function proxy(contextOrUri: string | string[] | proxy.Filter, config: proxy.Config): proxy.Proxy;
+declare function proxy(contextOrUri: string | string[] | proxy.Filter, config?: proxy.Config): proxy.Proxy;
 
 declare namespace proxy {
   type Filter = (pathanme: string, req: http.IncomingMessage) => boolean;
```
Comment (BendingBender) [inline types/http-proxy-middleware/index.d.ts]: I know you didn't change this, but yould you please fix the typo in `pathanme`?

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/http-proxy-middleware/index.d.ts (hunk 2 above)
Comment (CelineBen) [reply types/http-proxy-middleware/index.d.ts]: Yep, done !

**Review by BendingBender — APPROVED** [review]

File: types/http-proxy-middleware/index.d.ts (hunk 2 above)
Comment (BendingBender) [reply types/http-proxy-middleware/index.d.ts]: Thank you!

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**